doc/man3/OSSL_CMP_ITAV_set0.pod

```diff
@@ -0,0 +1,89 @@
+=pod
+
+=head1 NAME
+
+OSSL_CMP_ITAV_set0,
+OSSL_CMP_ITAV_get0_type,
+OSSL_CMP_ITAV_get0_value,
+OSSL_CMP_ITAV_stack_item_push0
+- OSSL_CMP_ITAV utility functions
+
+=head1 SYNOPSIS
+
+  #include <openssl/cmp.h>
+  void OSSL_CMP_ITAV_set0(OSSL_CMP_ITAV *itav, ASN1_OBJECT *type,
+                          ASN1_TYPE *value);
+  ASN1_OBJECT *OSSL_CMP_ITAV_get0_type(const OSSL_CMP_ITAV *itav);
+  ASN1_TYPE *OSSL_CMP_ITAV_get0_value(const OSSL_CMP_ITAV *itav);
+
+  int OSSL_CMP_ITAV_stack_item_push0(STACK_OF(OSSL_CMP_ITAV) **itav_sk_p,
+                                     OSSL_CMP_ITAV *itav);
+
+=head1 DESCRIPTION
+
+OSSL_CMP_ITAV_set0() sets the B<itav> with an infoType of B<type> and an
+infoValue of B<value>. This function uses the pointers B<type> and B<value>
+internally, so they must B<not> be freed up after the call.
+
+OSSL_CMP_ITAV_get0_type() returns a direct pointer to the infoType in the
+B<itav>.
+
+OSSL_CMP_ITAV_get0_value() returns a direct pointer to the infoValue in
+the B<itav> as generic ASN1_TYPE*.
+
+OSSL_CMP_ITAV_stack_item_push0() pushes B<itav> to the stack pointed to
+by B<*itav_sk_p>. It creates a new stack if B<*itav_sk_p> points to NULL.
+
+=head1 NOTES
+
+CMP is defined in RFC 4210 (and CRMF in RFC 4211).
+
+=head1 RETURN VALUES
+
+OSSL_CMP_ITAV_set0() does not return a value.
+
+OSSL_CMP_ITAV_stack_item_push0() returns 1 on success, 0 on error.
+
+=head1 EXAMPLE
+
+The following code creates and sets a structure representing a generic
+InfoTypeAndValue sequence, using an OID created from text as type, and an
+integer as value. Afterwards, it is pushed to the OSSL_CMP_CTX to be later
+included in the requests' PKIHeader's genInfo field.
+
+
+    ASN1_OBJECT *type = OBJ_txt2obj("1.2.3.4.5", 1);
+
+    ASN1_INTEGER *asn1int = ASN1_INTEGER_new();
+    ASN1_INTEGER_set(asn1int, 12345);
+
+    ASN1_TYPE *val = ASN1_TYPE_new();
+    ASN1_TYPE_set(val, V_ASN1_INTEGER, asn1int);
+
+    OSSL_CMP_ITAV *itav = OSSL_CMP_ITAV_new();
+    OSSL_CMP_ITAV_set0(itav, type, val);
+
+    OSSL_CMP_CTX *ctx = OSSL_CMP_CTX_new();
+    if (!OSSL_CMP_CTX_geninfo_itav_push0(ctx, itav)) {
+        OSSL_CMP_ITAV_free(itav);
+        goto err;
+    }
+
+    ...
+
+    OSSL_CMP_CTX_delete(cmp_ctx); /* also frees itav */
```

Comment:
Is this function from a future PR? By convention we normally call these things `*_free()`

Reply:
Yes, this will be part of chunk 4.
`OSSL_CMP_CTX_delete()` internally uses the generically defined `OSSL_CMP_CTX_free()`, and in addition does cleanup of non-ASN1 structures, so we cannot give it the usual name.

Reply:
Not sure I'm too keen on that. But since we've not seen the implementation of this yet, I guess I'm ok to defer the discussion until chunk 4 has arrived.

Reply:
All right, let's see - we'll find a solution...